crates/astria-sequencer-relayer/src/relayer/celestia_client/mod.rs

```diff
@@ -0,0 +1,654 @@
+    // event that a blob is too large.
+    let blob_sizes = blob_sizes
+        .into_iter()
+        .map(|blob_size| {
+            u32::try_from(blob_size).map_err(|_| Error::BlobTooLarge {
+                byte_count: blob_size,
+            })
+        })
+        .collect::<Result<_, _>>()?;
+
+    Ok(MsgPayForBlobs {
+        signer: signer.0,
+        namespaces,
+        blob_sizes,
+        share_commitments,
+        share_versions,
+    })
+}
+
+/// Extracts a `BaseAccount` from the given response.
+fn account_from_response(
+    response: Result<Response<QueryAccountResponse>, Status>,
+) -> Result<BaseAccount, Error> {
+    let account_info = response
+        .map_err(|status| Error::FailedToGetAccountInfo(GrpcResponseError::from(status)))?;
+
+    let account_as_any = account_info
+        .into_inner()
+        .account
+        .ok_or_else(|| Error::EmptyAccountInfo)?;
+    let expected_type_url = BaseAccount::type_url();
+
+    if expected_type_url == account_as_any.type_url {
+        return BaseAccount::decode(&*account_as_any.value)
+            .map_err(|error| Error::DecodeAccountInfo(ProtobufDecodeError::from(error)));
+    }
+
+    Err(Error::AccountInfoTypeMismatch {
+        expected: expected_type_url,
+        received: account_as_any.type_url,
+    })
+}
+
+/// Extracts the minimum gas price from the given response.
+fn min_gas_price_from_response(
+    response: Result<Response<MinGasPriceResponse>, Status>,
+) -> Result<f64, Error> {
+    const UNITS_SUFFIX: &str = "utia";
+    let min_gas_price_with_suffix = response
+        .map_err(|status| Error::FailedToGetMinGasPrice(GrpcResponseError::from(status)))?
+        .into_inner()
+        .minimum_gas_price;
+    let min_gas_price_str = min_gas_price_with_suffix
+        .strip_suffix(UNITS_SUFFIX)
+        .ok_or_else(|| Error::MinGasPriceBadSuffix {
+            min_gas_price: min_gas_price_with_suffix.clone(),
+            expected_suffix: UNITS_SUFFIX,
+        })?;
+    min_gas_price_str
+        .parse::<f64>()
+        .map_err(|source| Error::FailedToParseMinGasPrice {
+            min_gas_price: min_gas_price_str.to_string(),
+            source,
+        })
+}
+
+/// Extracts the tx hash from the given response.
+fn tx_hash_from_response(
+    response: Result<Response<BroadcastTxResponse>, Status>,
+) -> Result<TxHash, Error> {
+    let tx_response = response
+        .map_err(|status| Error::FailedToBroadcastTx(GrpcResponseError::from(status)))?
+        .into_inner()
+        .tx_response
+        .ok_or_else(|| Error::EmptyBroadcastTxResponse)?;
+    if tx_response.code != 0 {
+        let error = Error::BroadcastTxResponseErrorCode {
+            tx_hash: tx_response.txhash,
+            code: tx_response.code,
+            namespace: tx_response.codespace,
+            log: tx_response.raw_log,
+        };
+        return Err(error);
+    }
+    debug!(tx_hash = %tx_response.txhash, "broadcast transaction succeeded");
```

Comment:
```suggestion
    debug!(tx_hash = %tx_response.txhash, "transaction included in celestia block");
```

maybe log celestia block height if you can?